### ISSUE TITLE
Struggling to install Dafny on VS Code with MacOS: can't find dotnet

### ISSUE THREAD

**Issue by johnwickerson** [dafny-lang/dafny#1737]
Hi, I'm struggling to install Dafny on my new Macbook M1. I'm running MacOS Monterey (12.0). I've managed to get dafny running fine on the command line, but am struggling to get VS Code integration working. I've installed the dotnet SDK, and you can see in the screenshot below that `dotnet` is in the PATH that VS Code uses. But each time I open a `.dfy` file, I get 

> There is no compatible dotnet runtime installed. Dafny requires the ASP.NET Core Runtime 5.0.

I've tried various things in my `dafny.dotnetExecutablePath` in settings, but nothing seems to help. Any tips? Thanks very much.

<img width="652" alt="Screenshot 2022-01-20 at 14 14 11" src="https://user-images.githubusercontent.com/4182011/150355440-0efd8fdd-a219-4565-9c33-427879e25b37.png">

**Comment by fabiomadge** [comment]
Can you try installing `net5.0` as well? Support for `net6.0` will come with Dafny 3.4.

**Comment by johnwickerson** [comment]
Thanks @fabiomadge. With your tip, and quite a bit more effort, I've got it working! I'd better document my steps in case it's helpful to others running Mac OS Monterey on an M1 MacBook.

- I downloaded the .NET Runtime (MacOS, x64, installer) from https://dotnet.microsoft.com/en-us/download/dotnet/5.0. I'm not sure if that was actually necessary, as perhaps it was superseded by the next step.
- I downloaded the ASP.NET Core Runtime (MacOS, x64, binary) from the same page. I wasn't sure what to do with the downloaded .zip file as there was no installer, so I just unzipped it to `~/Applications/aspnetcore-runtime-5.0.13-osx-x64`.
- In VS Code settings (accessed via `CMD-,`) I set "Dafny: Dotnet Executable Path" to `~/Applications/aspnetcore-runtime-5.0.13-osx-x64/dotnet`. (Perhaps an alternative would have been to add that to my global `$PATH` environment variable.)
- VS Code is now clearly _trying_ to launch the right `dotnet` executable, but can't because MacOS doesn't trust the binary. So I went to `~/Applications/aspnetcore-runtime-5.0.13-osx-x64` and ran `xattr -d com.apple.quarantine dotnet` and `xattr -d com.apple.quarantine host/fxr/5.0.13/libhostfxr.dylib` to remove those checks.
- And it all works!
